pkgs/development/python-modules/torchvision/default.nix

```diff
@@ -1,30 +1,41 @@
-{ lib
-, symlinkJoin
-, buildPythonPackage
+{ buildPythonPackage
+, cudaSupport ? torch.cudaSupport or false # by default uses the value from torch
 , fetchFromGitHub
-, ninja
-, which
+, lib
 , libjpeg_turbo
 , libpng
+, ninja
 , numpy
-, scipy
 , pillow
-, torch
 , pytest
-, cudaSupport ? torch.cudaSupport or false # by default uses the value from torch
+, scipy
+, symlinkJoin
+, torch
+, which
 }:
 
 let
-  inherit (torch) gpuTargetString;
-  inherit (torch.cudaPackages) cudatoolkit cudnn;
+  inherit (torch) cudaPackages gpuTargetString;
+  inherit (cudaPackages) cudatoolkit cudaFlags cudaVersion;
 
-  cudatoolkit_joined = symlinkJoin {
-    name = "${cudatoolkit.name}-unsplit";
-    paths = [ cudatoolkit.out cudatoolkit.lib ];
+  cuda-redist = symlinkJoin {
```

Comment:
I just tried building this without `symlinkJoin` and it seems to work: we can just move `cuda_nvcc` over to `nativeBuildInputs` and the rest to `buildInputs`. This way if some of these libraries isn't actually used it won't end up in the final closure

Reply:
I don't know how I missed the difference between `nativeBuildInputs` and `buildInputs` previously -- what you said makes sense so I'll do that!